### PR TITLE
refactor: FORMS-965 hasSubmissionPermissions cleanup

### PR DESCRIPTION
**Pull request by WalterMoar**
# Description

Ideally we'd write tests and then refactor the `userAccess.hasSubmissionPermissions` middleware. However, it needs some changes that would break the tests, particularly with how error responses are created. Do a small refactor so that tests can be written:
- validate the `formSubmissionId` and 400 if it is missing or not a uuid
- `throw Problem` instead of `return next(Problem)` (code climate complaint about early returns)
- simplify interface by always taking an array of permissions (previously allowed a string as a shortcut to an array of one string)

Expand the tests:
- cover the new behaviour above
- increase test coverage
- organize the tests by response code / functionality
- update tests for `hasFormPermissions` to match new layout
- add new tests for `hasFormPermissions` to cover edge cases and service errors

## Types of changes

Refactor (improve code readability and consistency)
Tests (improve automated testing)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/src/forms/auth/middleware/userAccess.js [hunk 1]
```diff
@@ -200,23 +204,29 @@ const hasSubmissionPermissions = (permissions) => {
...
+      if (!submissionPermission) {
+        // no access to this submission...
+        throw new Problem(401, {
+          detail: 'You do not have access to this submission.',
+        });
+      }
 
-      // no access to this submission...
-      return next(new Problem(401, { detail: 'You do not have access to this submission.' }));
+      return next();
```
Comment (WalterMoar) [inline app/src/forms/auth/middleware/userAccess.js:229]: Just flipping the logic here so that the last thing that's done is call `next()`.

**Review by jasonchung1871 — APPROVED** [review]

**Comment by WalterMoar** [comment]
Closing this to clean up the number of PRs. Needs a massive amount of rework due to the SSO changes, but will reopen when there is time to work on it.
